remove ops in the __caffe2 namespace

bdhirsh opened the pull request:
Right now I'm removing ops in the `__caffe2` namespace out of convenience, as part of a refactor of the `export_caffe2_op_to_c10.` macro.

@linbinyu sorry for the tags in multiple places- I split up my PR into a few smaller diffs.

I git blame'd the __caffe2 ops and from [the PR here](https://github.com/pytorch/pytorch/commit/b704f30189a6a9f6551b0554ddc82ae5b9241e2c) it sounds like that separate namespace is required for the lite interpreter.  is that right that I shouldn't actually remove these ops? It'll just require a bit more refactoring.

Original PR (that is now split up) [is here](https://github.com/pytorch/pytorch/pull/46923)

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47322 make duplicate def() calls an error in the dispatcher
* #47321 migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API
* #47320 rename macro. TORCH_LIBRARY_FRAGMENT_THIS_API_IS_FOR_PER_OP_REGISTRATION_ONLY to TORCH_LIBRARY_FRAGMENT
* #47319 update legacy dispatcher registration API tests to avoid duplicate def() calls
* **#47318 remove ops in the __caffe2 namespace**

Differential Revision: [D24714807](https://our.internmc.facebook.com/intern/diff/D24714807)

bdhirsh committed:
remove ops in the __caffe2 namespace

[ghstack-poisoned]

ezyang commented:
@iseeyuan can you help review this?

iseeyuan commented:
@linbinyu , have we moved all maskrcnn ops from c2 to pt?

linbinyu commented:
export_caffe2_op_to_c10 is only used in server build but not mobile. Note that the "_caffe2" namespace is still there, this PR is to remove "_ _caffe2" so it's fine.

bdhirsh committed:
Update on "remove ops in the __caffe2 namespace"

Right now I'm removing ops in the `__caffe2` namespace out of convenience, as part of a refactor of the `export_caffe2_op_to_c10.` macro.

@linbinyu sorry for the tags in multiple places- I split up my PR into a few smaller diffs.

I git blame'd the __caffe2 ops and from [the PR here](https://github.com/pytorch/pytorch/commit/b704f30189a6a9f6551b0554ddc82ae5b9241e2c) it sounds like that separate namespace is required for the lite interpreter.  is that right that I shouldn't actually remove these ops? It'll just require a bit more refactoring.

Original PR (that is now split up) [is here](https://github.com/pytorch/pytorch/pull/46923)


Differential Revision: [D24714807](https://our.internmc.facebook.com/intern/diff/D24714807)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove ops in the __caffe2 namespace"

Right now I'm removing ops in the `__caffe2` namespace out of convenience, as part of a refactor of the `export_caffe2_op_to_c10.` macro.

@linbinyu sorry for the tags in multiple places- I split up my PR into a few smaller diffs.

I git blame'd the __caffe2 ops and from [the PR here](https://github.com/pytorch/pytorch/commit/b704f30189a6a9f6551b0554ddc82ae5b9241e2c) it sounds like that separate namespace is required for the lite interpreter.  is that right that I shouldn't actually remove these ops? It'll just require a bit more refactoring.

Original PR (that is now split up) [is here](https://github.com/pytorch/pytorch/pull/46923)


Differential Revision: [D24714807](https://our.internmc.facebook.com/intern/diff/D24714807)

[ghstack-poisoned]
